### PR TITLE
Add release instructions for IPython.

### PR DESCRIPTION
**Pull request by Carreau**
Mostly adapt from notebooks one.

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -21,6 +21,7 @@ Contents
    interactive/index
    config/index
    development/index
+   development/release_process
```
Comment (takluyver) [inline docs/source/index.rst]: I don't think this needs a separate entry in the top-level TOC - let's add it to `development/index` instead.

File: setup.py [hunk 2]
```diff
@@ -287,7 +287,14 @@ def run(self):
...
 setup_args.update(setuptools_extra_args)
 
+
+# loose as `.dev` is suppose to be invalid
+loose_pep440re = re.compile('^(\d+)\.(\d+)\.(\d+((a|b|rc)\d+)?)(\.post\d+)?(\.dev\d*)?$')
+
 def main():
+    import IPython.core.release as r
+    if not loose_pep440re.match(r.version):
+        raise ValueError("Version number '%s' is not valid (should match [N!]N(.N)*[{a|b|rc}N][.postN][.devN])" % r.version)
```
Comment (takluyver) [inline setup.py]: Do we want to do this check on every use of setup.py, or only when building packages for release?

File: docs/source/index.rst (hunk 1 above)
Comment (Carreau) [inline docs/source/index.rst]: I will move it.

File: setup.py (hunk 2 above)
Comment (Carreau) [inline setup.py]: Is it worth the added checking only on build ?

Comment (takluyver) [inline setup.py]: This is probably fine.

**Comment by Carreau** [comment]
comment addressed and rebased.

**Comment by Carreau** [comment]
WTF ? Why does not travis install traitlets ?

File: setup.py (hunk 2 above)
Comment (Carreau) [inline setup.py]: Change my mind, couldn't extract dependency from `setup.py` if traitlets not installed when you import `IPython.core.release` in `setup.py`

File: docs/source/development/index.rst [hunk 3]
```diff
@@ -26,3 +26,51 @@ on the IPython GitHub wiki.
    pycompat
    config
    inputhook_app
+
+Making an IPython release
+=========================
```
Comment (takluyver) [inline docs/source/development/index.rst]: I actually meant just adding the new page to the TOC in this file, not putting the content in here, but it's fine.

**Pull request merged**